### ISSUE TITLE
Implement garbage collecting of old issues

### ISSUE THREAD

**Issue by rashadg1030** [kowainik/issue-wanted#98]
This will require to fetch all closed issues and delete all closed issues from our database.

**Comment by rashadg1030** [comment]
I'm gonna use the `updated_at` field to determine if the issue should be deleted. What should time limit between updates be? If the issue isn't updated in 14 days or so or less?

**Comment by chshersh** [comment]
@rashadg1030 This issue is a bit more complicated. Open issues should never be deleted, no matter how old they are. _Old_ here means _is no longer open_. So you need to fetch all closed issues and delete all of them from our DB.

**Comment by rashadg1030** [comment]
@chshersh got it, make sense. Now how do I determine if an issue in the DB has closed or not? Do I just fetch all closed Haskell issues and delete the ones that match in the DB?

**Comment by chshersh** [comment]
> Do I just fetch all closed Haskell issues and delete the ones that match in the DB?

Yep, exactly. This should work for now. But architecture behind this implementation is not that straightforward. If you will repeatably call `deleteClosedIssues` and `upsertIssues` in `forever` loop, DB won't like it...

**Comment by rashadg1030** [comment]
@chshersh Can you explain why ` forever` is bad in this situation? I'm curious.

**Comment by chshersh** [comment]
Well, it's just bad in general for performance to do concurrently batch updates of different types and perform SELECT queries on a single time very fast. I'm not sure, but there's a chance that our sync can be very fast (like 5-20 minutes or even faster). Updating table with such frequency might be a high load for DB. But now I think that we don't need to do concurrent garbage collecting (this was my implicit assumption at first). The algorithm can be:

1. Upsert repos
2. Delete closed issues
3. Upsert open issues

**Comment by rashadg1030** [comment]
@chshersh Sounds good
